synapse/lib/stormtypes.py

```diff
@@ -2006,19 +2006,22 @@ async def _methQueueGet(self, offs=0, cull=True, wait=True):
 
         return await self.runt.dyncall('cortex', todo, gatekeys=gatekeys)
 
-    async def _methQueuePop(self, offs=0, cull=True, wait=True):
+    async def _methQueuePop(self, offs=None):
 
-        offs = await toint(offs)
-        wait = await toint(wait)
-
-        todo = s_common.todo('coreQueueGet', self.name, offs, cull=cull, wait=wait)
+        offs = await toint(offs, noneok=True)
         gatekeys = self._getGateKeys('get')
 
-        valu = await self.runt.dyncall('cortex', todo, gatekeys=gatekeys)
-        if valu is not None:
-            await self._methQueueCull(valu[0])
+        # emulate the old behavior on no argument
+        if offs is None:
+            todo = s_common.todo('coreQueueGets', self.name, 0, cull=True, wait=False)
```

Comment:
Using ``coreQueueGet`` avoids the need for the iterator

Reply:
But doesn't implement the backward compatible behavior.